bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java

```diff
@@ -307,6 +308,15 @@ public int run() throws Throwable {
             return ((IApp) c.newInstance()).run(this);
         }
 
+        // Explode war if necessary
+        if(FilenameUtils.getExtension(warDir.getAbsolutePath()).equals("war")) {
+            System.out.println("Exploding jenkins.war, this might take some time.");
+            String warPath = warDir.getAbsolutePath();
+            warDir = Util.explodeWar(warPath);
```

Comment:
would be better to use another variable so that CLI arguments remain immutable

Reply:
But `.collectJars(new File(warDir,"WEB-INF/lib"))` would need to pick up the WEB-INF file from the exploded war right, so we might need to change this new exploded file path global so it can be used by all functions or pass it as a parameter

Reply:
What's the problem without exploding the war file?

Reply:
> Currently --war option takes a part to the unzipped WAR file directory. Although it is good from the performance point of view, it causes confusion since Jenkinsfile Runner shows an odd behavior (classloading failures) when a WAR file is passed.
Refer: https://github.com/jenkinsci/jenkinsfile-runner/issues/224#issue-518539244